pkg/distro/defs/rhel-8/distro.yaml

```diff
@@ -875,14 +1187,33 @@ image_types:
                 include:
                   - "redhat-cloud-client-configuration"
 
-  ami:
-    <<: *ec2
-    package_sets:
-      os:
-        - *ec2_common_pkgset
-
   "ec2-sap":
     <<: *ec2
+    # XXX: this is ugly, YAML only supports shallow merging of maps so
+    # we we want to merge e.g. the version conditions that sap and ec2
+    # have we need to manually "unroll" it. we need to think of a better
+    # way here.
```

Comment:
I agree that we need a better way to do this.

I think that if we could resolve the conditions before merging image configs, it would be easier. This would most probably mean having a special **key** with a list of image configs to inherit from, similar to what we used to do in the code. This would mean not doing all the merging in YAML. Trying to merge data structures with conditions makes this unnecessarily complex.

Reply:
This becomes a lot nicer with PR#1600, c.f. https://github.com/osbuild/images/compare/main...mvo5:yaml-imageconfig-rhel8-starlark?expand=1#diff-c8017947ffd600efd01b571e33783872a8ae5eec8e7285b7563b6e567394e1ccR1074 which is this PR ported to this new mechanism (still not perfect but a lot better IMHO).

Reply:
This is now improved via pr#1608